README.md

```diff
@@ -1,94 +1,60 @@
-
-    ```elixir
-    def deps do
-      [{:ok, "~> 1.5.0"}]
-    end
-    ```
-
-## Usage
-
-The erlang convention for functions that can fail is to return a result tuple.
-A result tuple is a two-tuple tagged either as a success(`:ok`) or a failure(`:error`).
+- [Install from Hex](https://hex.pm/packages/ok)
+- [Documentation available on hexdoc](https://hexdocs.pm/ok)
 
 The OK module works with result tuples by treating them as a result monad.
 
 ```elixir
 {:ok, value} | {:error, reason}
 ```
 
-[Forum discussion on :ok/:error](https://elixirforum.com/t/usage-of-ok-result-error-vs-some-result-none/3253)
+See [Handling Errors in Elixir](http://insights.workshop14.io/2015/10/18/handling-errors-in-elixir-no-one-say-monad.html) for a more detailed explanation.
 
-### Result pipelines `~>>`
+### `OK.with`
 
-This macro allows pipelining result tuples through a pipeline of functions.
-The `~>>` macro is the is equivalent to bind/flat_map in other languages.
+`OK.with` allows for more concise and ultimately more readable code than the native `with` construct. It does this by leveraging result monads for both the happy and non-happy paths. By extracting the actual function return values from the result tuples, `OK.with` reduces noise which improves readability and recovers precious horizontal code real estate. This also encourages writing idiomatic Elixir functions which return `:ok`/`:error` tuples.
 
-```elixir
-import OK, only: ["~>>": 2]
+* [Elegant error handling with result monads, alternative to elixir `with` special form](https://elixirforum.com/t/elegant-error-handling-with-result-monads-alternative-to-elixir-with-special-form/3264/1)
+* [Discussion on :ok/:error](https://elixirforum.com/t/usage-of-ok-result-error-vs-some-result-none/3253)
 
-def get_employee_data(file, name) do
-  {:ok, file}
-  ~>> File.read
-  ~>> Poison.decode
-  ~>> Dict.fetch(name)
-end
-
-def handle_user_data({:ok, data}), do: IO.puts("Contact at #{data["email"]}")
-def handle_user_data({:error, :enoent}), do: IO.puts("File not found")
-def handle_user_data({:error, {:invalid, _}}), do: IO.puts("Invalid JSON")
-def handle_user_data({:error, :key_not_found}), do: IO.puts("Could not find employee")
-
-get_employee_data("my_company/employees.json")
-|> handle_user_data
-```
-
-Code structured like this is an example of [railway programming](http://www.zohaib.me/railway-programming-pattern-in-elixir/).
-
-[Forum discussion on error handling in pipelines](https://elixirforum.com/t/ok-elegant-error-handling-for-elixir-pipelines-version-1-0-released/1932)
-
-### Result blocks `with`
+#### Basic Usage
 
-For situations when the pipeline macro is not sufficiently flexible.
+* Use the `<-` operator to match & extract a value for an `:ok` tuple.
+* Use the `=` operator as you normally would for pattern matching an untagged result.
+* The last line should either be a function that returns the tuple, or the literal tuple itself.
```

Comment:
3 -  Return a result, Must also be in the form of a tagged tuple.
4 - *Optionally* handle some of the errors in an else block

```elixir
 save_order(order)
else
  :user_not_found ->
    {:error, :unauthorized} # Don't give away if users exist or not in your system
```